src/main/java/com/unb/tracker/validator/CourseValidator.java

```diff
@@ -21,11 +22,35 @@ public boolean supports(Class<?> aClass) {
     public void validate(Object o, Errors errors) {
         Course course = (Course) o;
 
-        if (course.getRows() < 1 || course.getRows() > 18) {
+        ValidationUtils.rejectIfEmptyOrWhitespace(errors,"name", "Ensure all required fields are filled. (Check Course Name)");
+        ValidationUtils.rejectIfEmptyOrWhitespace(errors,"rows", "Ensure all required fields are filled. (Check Rows)");
+        ValidationUtils.rejectIfEmptyOrWhitespace(errors,"cols", "Ensure all required fields are filled. (Check Columns)");
+
+        if (!courseService.findByInstructorUsernameAndNameAndSection(course.getInstructor().getUsername(), course.getName(), course.getSection()).isEmpty()) {
+            errors.rejectValue("name", "This section for this course already exists.");
+        }
+
+        if (course.getStartDate().toString().equals("1970-01-01")) {
+            errors.rejectValue("startDate", "Ensure all required fields are filled. (Check Start Date)");
+        }
+
+        if (course.getName().length() > 10) {
+            errors.rejectValue("name", "Course name should be less than 10 characters");
```

Comment:
Have we decided that we are going to limit the course name and section name to 10 characters? I still think we should just truncate with CSS, but if PO @mwalz1 wants this, I'll live ;)

Reply:
This was specified by MacIsaac